File: 백준/Silver/18110. solved．ac/solved．ac.py
import sys
input = sys.stdin.readline

N = int(input())
lst = [int(input()) for _ in range(N)]
if not N:
    print(0)
else:
    lst.sort()
    num = round((N*0.15)+0.0000000000001)
    print(round((sum(lst[num:N-num])/(N - 2*num))+0.0000000000001))

File: 백준/Silver/18110. solved．ac/solved．ac.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
let input = fs.readFileSync(filePath).toString().trim().split('\n');

const N = Number(input.shift())
const lst = []
input.forEach((num) => lst.push(Number(num)))

if (N == 0) {
    console.log(0)
}
else {
    lst.sort((a, b) => a-b)
    const num = Math.round(N*0.15)
    let totSum = 0;
    for (let i = num; i < N-num; i++) {
        totSum += lst[i]
    }
    console.log(Math.round(totSum/(N-2*num)))
}

